rcl/test/rcl/test_arguments.cpp

```diff
@@ -194,6 +167,56 @@ TEST_F(CLASSNAME(TestArgumentsFixture, RMW_IMPLEMENTATION), check_known_vs_unkno
   EXPECT_FALSE(are_known_ros_args({"--ros-args", "__log_level:=foo"}));
 }
 
+bool
+are_valid_ros_args(std::vector<const char *> argv)
```

Comment:
In all the cases you used `are_valid_ros_args`, you could have used `are_known_ros_args`. It does the same checking and also checks that you don't have unparsed args.

Reply:
That's true of the cases that are being tested here. Unknown ROS args (to `rcl`) are still valid ROS args.  I've added a asserts to check for this distinction.